src/groupWith.js

```diff
@@ -0,0 +1,35 @@
+var _curry2 = require('./internal/_curry2');
+
+/**
+ * Takes a list and returns a list of lists where each sublist contains only equal elements.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig (a, a -> Boolean) -> [a] -> [[a]]
+ * @param {Function} fn Function :: a, a -> Boolean
+ * @param {Array} list The array to group
+ * @return {List} A list that contains sublists of equal elements,
+ *         whose concatenations is equal to the original list.
+ * @example
+ *
+ *      var groupWithLetters = R.groupWith(function(letter1, letter2) {
+ *        return letter1 === letter2;
+ *      });
+ *      var letters = ["M", "i", "s", "s", "i", "s", "s", "i", "p", "p", "i"];
+ *      groupWithLetters(letters);
+ *      // ["M","i","ss","i","ss","i","pp","i"]
+ */
+module.exports = _curry2(function(fn, list) {
+  var result = [];
+  var idx = 0;
+  while (idx < list.length) {
+    var nextidx = idx;
```

Comment:
We could, although we'd have to remember to push `list[idx]`onto the resultant array in that case though.